### ISSUE TITLE
Typescript types are broken when using a `webpack.config.ts` file

### ISSUE THREAD

**Issue by PodaruDragos** [webpack/webpack-dev-server#4181]
### Bug report
When using `webpack-dev-server:4.7.3`

### Actual Behavior
```
any
Property 'devServer' does not exist on type 'Configuration'.ts(2339)
```

### Expected Behavior
Typescript should not see any issues here, just like it worked before when we had types from `@types/webpack-dev-server`

### How Do We Reproduce?

Install 
```
    "webpack": "5.66.0",
    "webpack-cli": "4.9.1",
    "webpack-dev-server": "4.7.3"
```

add a `webpack.config.ts` file.

```
import { Configuration ,} from 'webpack';
const configuration: Configuration = {};
configguration.devServer = { 

};

Error should be there.
```

**Comment by alexander-akait** [comment]
You should import webpack dev server types, answer https://github.com/webpack/webpack/discussions/15076#discussioncomment-1887720

**Comment by Trinovantes** [comment]
Typescript by default makes every definition in `node_modules/@types` global. The old `@types/webpack-dev-server` package has a side effect of extending `webpack.Configuration` object which means everything "just works" after installing `webpack-dev-server` and `@types/webpack-dev-server`

Now that the types are included in 4.7, the side effects cannot be applied globally without modifying the `types` property in `tsconfig.json`

Instead I was able to fix the missing types by adding a side-effect import:
```ts
import { Configuration } from 'webpack'
import 'webpack-dev-server' // Modifies the Configuration definition

const config: Configuration = {
    devSever: {}
}

export default config
```

**Comment by fregante** [comment]
Since my tsconfig has an explicit `compilerOptions.types` definition, it only started working after specifying in it:

```jsonc
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    // etc
    "types": [
      // etc
      "webpack-dev-server"
    ]
  }
}
```
